ext/drx/drx.c

```diff
@@ -2836,8 +2929,251 @@ static bool
+            if (val == 0 || (val & (val - 1)) != 0)
+                break;
+            opnd_t dst0 = instr_get_dst(&params->inst, 0);
+            if (opnd_is_reg(dst0)) {
+                reg_id_t tmp_gpr = opnd_get_reg(dst0);
+                if (reg_is_gpr(tmp_gpr)) {
+                    params->gpr_bit_mask = tmp_gpr;
+                    advance_state(DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_6,
+                                  params);
+                    break;
+                }
+            }
+        }
+        skip_unknown_instr_inc(DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_0, params);
+        break;
+    }
+    case DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_6:
+        if (instr_get_opcode(&params->inst) == OP_kmovw) {
+            opnd_t src0 = instr_get_src(&params->inst, 0);
+            if (opnd_is_reg(src0) && opnd_get_reg(src0) == DR_REG_K0) {
+                opnd_t dst0 = instr_get_dst(&params->inst, 0);
+                if (opnd_is_reg(dst0)) {
+                    reg_id_t tmp_gpr = opnd_get_reg(dst0);
+                    if (reg_is_gpr(tmp_gpr)) {
+                        params->gpr_save_scratch_mask = tmp_gpr;
+                        advance_state(DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_7,
+                                      params);
+                        break;
+                    }
+                }
+            }
+        }
+        skip_unknown_instr_inc(DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_0, params);
+        break;
+    case DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_7:
+        ASSERT(params->gpr_bit_mask != DR_REG_NULL,
+               "internal error: expected gpr register to be recorded in state "
+               "machine.");
+        if (instr_get_opcode(&params->inst) == OP_kmovw) {
+            opnd_t src0 = instr_get_src(&params->inst, 0);
+            if (opnd_is_reg(src0) && opnd_get_reg(src0) == params->gpr_bit_mask) {
+                opnd_t dst0 = instr_get_dst(&params->inst, 0);
+                if (opnd_is_reg(dst0) && opnd_get_reg(dst0) == DR_REG_K0) {
+                    params->restore_scratch_mask_start_pc = params->pc;
+                    advance_state(DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_8,
+                                  params);
+                    break;
+                }
+            }
+        }
+        skip_unknown_instr_inc(DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_0, params);
+        break;
+    case DRX_DETECT_RESTORE_AVX512_SCATTER_EVENT_STATE_8:
+        if (instr_get_opcode(&params->inst) == OP_kandnw) {
+            opnd_t src0 = instr_get_src(&params->inst, 0);
+            opnd_t src1 = instr_get_src(&params->inst, 1);
+            opnd_t dst0 = instr_get_dst(&params->inst, 0);
+            if (opnd_is_reg(src0) && opnd_get_reg(src0) == DR_REG_K0) {
+                if (opnd_is_reg(src1) &&
+                    opnd_get_reg(src1) == params->sg_info->mask_reg) {
+                    if (opnd_is_reg(dst0) &&
```

Comment:
Another case of unnecessary indentation.

Reply:
Done.